app/src/main/java/fr/neamar/kiss/dataprovider/simpleprovider/CalculatorProvider.java

```diff
@@ -0,0 +1,83 @@
+package fr.neamar.kiss.dataprovider.simpleprovider;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import fr.neamar.kiss.dataprovider.IProvider;
+import fr.neamar.kiss.pojo.Pojo;
+import fr.neamar.kiss.pojo.SearchPojo;
+import fr.neamar.kiss.searcher.Searcher;
+
+public class CalculatorProvider implements IProvider {
+    private Pattern p;
+
+    public CalculatorProvider() {
+        p = Pattern.compile("([0-9.]+)\\s?([+\\-*/×÷])\\s?([0-9.]+)");
+    }
+
+    @Override
+    public void requestResults(String query, Searcher searcher) {
+        // Now create matcher object.
+        Matcher m = p.matcher(query);
+        if (m.find()) {
+            SearchPojo pojo = new SearchPojo();
+            pojo.id = "calculator://";
+            pojo.type = SearchPojo.CALCULATOR_QUERY;
+
+            String operator = m.group(2);
+            String result = "";
+            // let's go for floating point arithmetic
+            float lhs = Float.parseFloat(m.group(1));
+            float rhs = Float.parseFloat(m.group(3));
+
+            float floatResult = 0;
+            if (operator.equals("+")) {
```

Comment:
Hey just thinking, if using switch case would be better suited? though this still works 😃